### PR TITLE
feat: #4759 venus-auth: add docker feature

### PR DESCRIPTION
**Pull request by LinZexiao**
## Related Issues
https://github.com/filecoin-project/venus/issues/4759

## Proposed Changes
add docker features
add github action for automaticly pushing to dockerhub

## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [x] The PR title is in the form of of `<PR type>: <#issue number> <area>: <change being made>`
    - example: ` fix: #1234 mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_, _misc_, _perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _venus_, _venus-messager_, _venus-miner_, _venus-gateway_, _venus-auth_, _venus-market_, _venus-sealer_, _venus-wallet_, _venus-cluster_, _api_, _chain_, _state_, _vm_, _data transfer_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_
- [x] This PR has tests for new functionality or change in behaviour
- [x] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in [venus docs](https://venus.filecoin.io) or [Discussion Tutorials.](https://github.com/filecoin-project/venus/discussions/categories/tutorials)
- [x] CI is green

### CODE REVIEW

File: .github/workflows/build_upload.yml [hunk 1]
```diff
@@ -174,4 +174,13 @@ jobs:
...
+          set -e
+
+      - name: Publish the Docker image 
+        if: ${{ github.ref_type == 'tag' && github.event_name == 'create' }}
+        run: |
+          docker build . --file dockerfile --tag filvenus/venus-auth:latest
+          docker tag filvenus/venus-auth:latest filvenus/venus-auth:${{steps.vars.outputs.github_tag }}
+          docker login --username=filvenus --password ${{ secrets.DOCKER_PASSWORD }} 
+          docker push filvenus/venus-auth:${{steps.vars.outputs.github_tag }}
+          docker push filvenus/venus-auth:latest
```
Comment (diwufeiwen) [inline .github/workflows/build_upload.yml:186]: 文件末空行

File: .github/workflows/docker-release.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Docker Image CI
+
+on:
+  create:
+    tags:
+      - '**'
+  push:
+    branches: [ docker ]
```
Comment (diwufeiwen) [inline .github/workflows/docker-release.yaml:8]: 这里是不是主分支`[ main ]`？

Comment (LinZexiao) [reply .github/workflows/docker-release.yaml:8]: 对，这个是只监听docker分支的，主分支的已经写进原来那个统一的那个 工作流文件了。如果以后docker分支不保留的话，可以删掉，否则建议保留，这样可以方便dockers分支进行发布镜像测试

File: .github/workflows/build_upload.yml (hunk 1 above)
Comment (LinZexiao) [reply .github/workflows/build_upload.yml:186]: 好的，我会在文件补上空行

**Pull request closed without merging**